File: virtual-world-simulator/src/Entity/Plants/Grass.cpp
#include "Grass.h"

Grass::Grass(World* world, Point position) :
	Plant { AssetManager::getAssetManager()->getAsset("grass"), world, position, 'G', 0, SpreadChance::MEDIUM }
{

}

void Grass::update(std::fstream& logFile)
{
	this->spread<Grass>(logFile);
	Entity::update(logFile);
}

File: Java/virtual-world-simulator/src/Entities/Plants/Grass.java
package Entities.Plants;

import App.World;
import Common.Point;

import javax.swing.*;

public class Grass extends Plant
{
    public Grass(World world, Point position)
    {
        super(new ImageIcon("assets/grass.png"), world, position, 'G', 0, 0, SpreadChance.MEDIUM);
    }

    public void Update()
    {
        super.Update();
        Spread(new Grass(this.world, new Point(-1, -1)));
    }
}
